pkg/v1/tkg/client/kubevip_update_helper.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	capikubeadmv1beta1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/clusterclient"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/constants"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
+)
+
+const (
+	vipLeaseDurationKey     = "VIP_LEASEDURATION"
+	vipRenewDeadlineKey     = "VIP_RENEWDEADLINE"
+	vipRetryPeriodKey       = "VIP_RETRYPERIOD"
+	vipLeaseDuration        = "vip_leaseduration"
+	vipRenewDeadline        = "vip_renewdeadline"
+	vipRetryPeriod          = "vip_retryperiod"
+	defaultLeaseDuration    = "15"
+	defaultRenewDeadline    = "10"
+	defaultRetryPeriod      = "2"
+	defaultNewLeaseDuration = "30"
+	defaultNewRenewDeadline = "20"
+	defaultNewRetryPeriod   = "4"
+)
+
+func (c *TkgClient) increaseKubeVipTimeouts(regionalClusterClient clusterclient.Client, upgradeClusterConfig *clusterUpgradeInfo) error {
+	currentKCP, err := regionalClusterClient.GetKCPObjectForCluster(upgradeClusterConfig.ClusterName, upgradeClusterConfig.ClusterNamespace)
+	if err != nil {
+		return errors.Wrapf(err, "unable to get KCP object to increase the kube-vip timeouts. Continuing upgrade with old parameters. ")
+	}
+	// If iaas != vsphere, skip trying to update KCP object
+	if currentKCP.Spec.MachineTemplate.InfrastructureRef.Kind != constants.VSphereMachineTemplate {
+		return nil
+	}
+
+	newKCP := currentKCP.DeepCopy()
+	var currentKubeVipPod corev1.Pod
+
+	for i := range currentKCP.Spec.KubeadmConfigSpec.Files {
+		err := yaml.Unmarshal([]byte(currentKCP.Spec.KubeadmConfigSpec.Files[i].Content), &currentKubeVipPod)
+		if err == nil {
```

Comment:
Should we check the name of the pod as well?
```suggestion
		if err == nil && currentKubeVipPod.GetName() == "kube-vip" {
```

Reply:
Will the name of the pod always just be `kube-vip` ? If yes, then this is a good idea

Reply:
Update: Added the name check as well